src/pubsub/mod_pubsub.erl

```diff
@@ -452,33 +464,18 @@ delete_pep_iq_handlers(ServerHost) ->
     gen_iq_handler:remove_iq_handler(ejabberd_sm, ServerHost, ?NS_PUBSUB),
     gen_iq_handler:remove_iq_handler(ejabberd_sm, ServerHost, ?NS_PUBSUB_OWNER).
 
-init_send_loop(ServerHost) ->
-    Plugins = gen_mod:get_module_opt(ServerHost, ?MODULE, plugins),
-    init_send_loop(ServerHost, gen_mod:get_module_opts(ServerHost, ?MODULE), Plugins).
-
-init_send_loop(ServerHost, #{last_item_cache := LastItemCache, max_items_node := MaxItemsNode,
-                             pep_mapping := PepMapping, ignore_pep_from_offline := PepOffline,
-                             access_createnode := Access},
-               Plugins) ->
+init_state(ServerHost, #{last_item_cache := LastItemCache, max_items_node := MaxItemsNode,
+                         pep_mapping := PepMapping, ignore_pep_from_offline := PepOffline,
+                         access_createnode := Access, plugins := Plugins}, Plugins) ->
```

Comment:
not really, I just does not understand the code 100% :) could change to the old behaviour though

Reply:
So maybe change remove the `plugins := Plugins` to avoid this unexpected pattern-matching?```

Reply:
tricky, there are these gen_server calls to get plugins (see plugins/1 function) instead of calling opts instead.
And gen_server is a pretty heavy operation.

Is there an use case for a plugin failing to start?
Maybe we could crash the module start function if the plugin crashes instead?

Reply:
For me this is out of scope of this PR - so why not keep the functionality as it was before?